web/src/components/layout/layout.tsx

```diff
@@ -194,6 +272,9 @@ class Layout extends React.PureComponent<LayoutProps, LayoutState> {
             teamToken={challengeTeamToken}
           />
         )}
+        {targetSegment && !localStorage.getItem(targetSegment.storageKey) && (
```

Comment:
Commented on this below, but we might wanna use this `useStickyState` hook: https://joshwcomeau.com/react/persisting-react-state-in-localstorage/

Reply:
I added the custom hook to our stores and also to `notification-banner`, but I'm not going to bother with Layout - it's a class component and it seemed kind of overkill to wrap it in an HOC just for this (esp since it's already wrapped in `withRouter`).